app/src/main/java/com/kamron/pogoiv/MainActivity.java

```diff
@@ -116,6 +118,7 @@
     private int pokemonHP;
     private boolean pokeFlyRunning = false;
     private int trainerLevel;
+    private LruCache<String, String> OCRCache = new LruCache<>(200);
```

Comment:
Re generated equals/hashCode, since we're using Lombok already we can avoid it: http://jnb.ociweb.com/jnb/jnbJan2010.html#equals

> Well, I won't want to hash them all together. If the same HP occurs in another pokemon, it might still get a cache hit.

I'm proposing to use `HashMap<Key, String>`; that way, looking up an HP hash can't find a Candy result, exactly like in your code using strings.